lib/dogma/util/ast_node.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Dogma.Util.ASTNode do
+  @moduledoc"""
+  Utility functions for analyzing and categorizing AST nodes
+  """
+
+  @doc """
+  Returns true if the node x is a literal
+      
+      iex> Dogma.Util.ASTNode.literal?( "foo" )
+      true
+
+      iex> Dogma.Util.ASTNode.literal?({:foo, [], Elixir})
+      false
+  """
+  def literal?(x) when is_atom    x do true end
+  def literal?(x) when is_binary  x do true end
+  def literal?(x) when is_boolean x do true end
+  def literal?(x) when is_list    x do true end
+  def literal?(x) when is_number  x do true end
+  def literal?({_,_})               do true end
+  def literal?({:{}, _, _})         do true end
+  def literal?({:%{}, _, _})        do true end
+  def literal?(node) do
+    sigil?(node)
+  end
+
+  def sigil?({fun, _, _}) when is_atom fun do
```

Comment:
Ooh, smart! Does it make sense to have macros for defining functions that match different classes of AST node? e.g.
```
def_for_sigils literal? ({sigil,_,_}) do false end
```
which would expand to define that function for all sigils?